backend/infrahub/pools/prefix.py

```diff
@@ -1,174 +1,220 @@
 from __future__ import annotations
 
 import ipaddress
-from collections import OrderedDict, defaultdict
-from ipaddress import IPv4Network, IPv6Network
-from typing import Optional, Union
+from typing import TYPE_CHECKING, Literal, Union
 
+from netaddr import IPSet
 
-class PrefixPool:
-    """
-    Class to automatically manage Prefixes and help to carve out sub-prefixes
+if TYPE_CHECKING:
+    from infrahub.core.ipam.constants import IPNetworkType
+
+
+def get_prefix_type(prefix_str: str) -> Literal["ipv4", "ipv6"]:
+    """Determines the type of prefix (ipv4 or ipv6) given a prefix string."""
+    if isinstance(ipaddress.ip_network(prefix_str), ipaddress.IPv4Network):
+        return "ipv4"
+    if isinstance(ipaddress.ip_network(prefix_str), ipaddress.IPv6Network):
+        return "ipv6"
+
+    raise ValueError(f"'{prefix_str}' is not a valid IPv4 or IPv6 Prefix")
```

Comment:
Would it be possible to return an integer for the IP version of the prefix? It would simplify this function (and be more accurate IMO) by having just:

```python
return ipaddress.ip_network(prefix_str).version
```

or let it raise an exception.